lib/openshift/cluster_logging.rb

```diff
@@ -23,15 +23,38 @@ class ClusterLogging < ClusterResource
       raw_resource(user: user, cached: cached, quiet: quiet).dig('status', 'visualization')
     end
 
+    private def collection_status_raw(user: nil, cached: true, quiet: false)
+      collection_raw(user: user, cached: cached, quiet: quiet)['logs']
+    end
+
+    private def rsyslog_status_raw(user: nil, cached: true, quiet: false)
+      collection_status_raw(user: user, cached: cached, quiet: quiet)['rsyslogStatus']
+    end
 
     private def fluentd_status_raw(user: nil, cached: true, quiet: false)
-      collection_raw(user: user, cached: cached, quiet: quiet).dig('logs', 'fluentdStatus')
+      collection_status_raw(user: user, cached: cached, quiet: quiet)['fluentdStatus']
     end
 
     private def curator_status_raw(user: nil, cached: true, quiet: false)
       self.curation(user: user, cached: cached, quiet: quiet)['curatorStatus']
     end
 
+    private def rsyslog_pods(user: nil, cached: true, quiet: false)
+      rsyslog_status_raw(user: user, cached: cached, quiet: quiet)['pods']
+    end
+
+    private def rsyslog_ready_pods(user: nil, cached: true, quiet: false)
+      rsyslog_pods(user: user, cached: cached, quiet: quiet)['ready']
+    end
+
+    private def rsyslog_failed_pods(user: nil, cached: true, quiet: false)
+      rsyslog_pods(user: user, cached: cached, quiet: quiet)['failed']
+    end
+
+    private def rsyslog_notready_pods(user: nil, cached: true, quiet: false)
+      rsyslog_pods(user: user, cached: cached, quiet: quiet)['notReady']
```

Comment:
for the methods above, better create `Pod` objects and return them instead of returning raw hash. Given methods are `privete` though, we can keep as is also.

Reply:
The output here is in json/yaml struct, and I think it's hard to return `Pod` objects. To check the rsyslog or fluentd pods ready or not, we use` rsyslog_nodes == rsyslog_ready_pods && rsyslog_failed_pods.count == 0 && rsyslog_notready_pods.count == 0`, so we only need the `count` here.

Example of status info:
```  
  status:
    collection:
      logs:
        fluentdStatus:
          daemonSet: fluentd
          nodes:
            fluentd-7qqdw: ip-10-0-162-127.eu-west-3.compute.internal
            fluentd-554s2: ip-10-0-158-63.eu-west-3.compute.internal
            fluentd-bd86w: ip-10-0-134-228.eu-west-3.compute.internal
            fluentd-m5kq4: ip-10-0-165-224.eu-west-3.compute.internal
            fluentd-mqt4h: ip-10-0-128-209.eu-west-3.compute.internal
            fluentd-wvr6f: ip-10-0-156-139.eu-west-3.compute.internal
          pods:
            failed: []
            notReady: []
            ready:
            - fluentd-554s2
            - fluentd-7qqdw
            - fluentd-bd86w
            - fluentd-m5kq4
            - fluentd-mqt4h
            - fluentd-wvr6f
```

Reply:
It is not hard to return `Pod` objects. It will be something like:
```
rsyslog_pods(user: user, cached: cached, quiet: quiet)['notReady'].map { |name|
  Pod.new(name: name, project: project)
}
```

But if we don't need the pods, then you can rename the methods to
```
rsyslog_notready_pod_names
rsyslog_ready_pod_names
...
```

Reply:
For the above suggestion to work, you'd have to fix how `ClusterLogging` resource is defined. It is presently `ClusterLogging < ClusterResource` but it is in fact a namespaces resource (it is within a namespace), thus it needs to be `ClusterLogging < ProjectResource`. @pruan-rht, please watch this in case anything relies on the wrong inheritance.

Reply:
YAML as a reference http://pastebin.test.redhat.com/799528